golang/cosmos/daemon/cmd/root.go

```diff
@@ -443,3 +443,61 @@ func (ac appCreator) appExport(
 
 	return gaiaApp.ExportAppStateAndValidators(forZeroHeight, jailAllowedAddrs)
 }
+
+// replaceCosmosSnapshotExportCommand monkey-patches the "snapshots export" command
+// added by cosmos-sdk and replaces its implementation with one suitable for
+// our modifications to the cosmos snapshots process
+func replaceCosmosSnapshotExportCommand(cmd *cobra.Command, ac appCreator) {
+	replacedRunE := func(cmd *cobra.Command, args []string) error {
+		ctx := server.GetServerContextFromCmd(cmd)
+
+		height, err := cmd.Flags().GetInt64("height")
+		if err != nil {
+			return err
+		}
+
+		home := ctx.Config.RootDir
+		dataDir := filepath.Join(home, "data")
+		db, err := dbm.NewDB("application", server.GetAppDBBackend(ctx.Viper), dataDir)
+		if err != nil {
+			return err
+		}
+
+		app := ac.newSnapshotsApp(ctx.Logger, db, nil, ctx.Viper)
+		gaiaApp := app.(*gaia.GaiaApp)
+
+		if height == 0 {
+			height = app.CommitMultiStore().LastCommitID().Version
+		}
+
+		cmd.Printf("Exporting snapshot for height %d\n", height)
```

Comment:
Should this output go to stderr?  Ideally, please use the `log.Logger`.

Reply:
I copied this part from the original which used `cmd.Printf`